api/app/src/providers/withings.ts

```diff
@@ -303,6 +303,10 @@ export class Withings extends Provider implements OAuth2 {
       params,
     });
 
+    if (response.data.error) {
+      throw new Error(response.data.error);
```

Comment:
We use Sentry to capture detailed errors, let's do the same here.

See [this](https://github.com/metriport/metriport/blob/face1d57b6dd4b00e112052149dda4f77305d972/api/app/src/providers/google.ts#L132-L137) other provider code for an example.